src/olympia/devhub/decorators.py

```diff
@@ -80,6 +92,30 @@ def fun():
         return decorator
 
 
+def readonly_if_site_permission(f):
+    """If the add-on is a site permission add-on, only allow readonly requests.
+
+    Add this decorator on top of dev_required() (which would normally allow
+    editing site permission add-ons) to disallow edition of specific devhub
+    pages for site permission add-ons."""
```

Comment:
maybe we still need this for some views?  But I don't think we should require it to be used too for any `@dev_required` decorated view.

Reply:
That's one of the reasons I went with the separate decorator route, but actually, that turned out to be false. So yeah, I agree, no good reason for a separate decorator.